mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1245,6 +1249,1321 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
+
+    auto dilationH =
+        dilationsAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto dilationW =
+        dilationsAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    auto strideH =
+        stridesAttr.getValue()[0].template dyn_cast<IntegerAttr>().getInt();
+    auto strideW =
+        stridesAttr.getValue()[1].template dyn_cast<IntegerAttr>().getInt();
+    // get y, x, ho, wo, hi, wi
+    int64_t g, n, k, c, y, x, ho, wo, hi, wi;
+    g = n = k = c = y = x = ho = wo = hi = wi = 0;
+    for (unsigned i = 0; i < filterLayoutAttr.size(); ++i) {
+      auto filterAttr =
+          filterLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+      auto inputAttr =
+          inputLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+      auto outputAttr =
+          outputLayoutAttr.getValue()[i].template dyn_cast<StringAttr>();
+
+      if (filterAttr.getValue() == "g") {
+        g = filterShape[i];
+      } else if (filterAttr.getValue() == "k") {
+        k = filterShape[i];
+      } else if (filterAttr.getValue() == "c") {
+        c = filterShape[i];
+      } else if (filterAttr.getValue() == "y") {
+        y = filterShape[i];
+      } else if (filterAttr.getValue() == "x") {
+        x = filterShape[i];
+      }
+
+      if (inputAttr.getValue() == "ni") {
+        n = inputShape[i];
+      } else if (inputAttr.getValue() == "hi") {
+        hi = inputShape[i];
+      } else if (inputAttr.getValue() == "wi") {
+        wi = inputShape[i];
+      }
+
+      if (outputAttr.getValue() == "ho") {
+        ho = outputShape[i];
+      } else if (outputAttr.getValue() == "wo") {
+        wo = outputShape[i];
+      }
+    }
+
+    // compute padding hi/wi.
+    auto hiPadded = hi + leftPadH + rightPadH;
+    auto wiPadded = wi + leftPadW + rightPadW;
+
+    auto gcdStrideDilationH = math::gcd(strideH, dilationH);
+    auto gcdStrideDilationW = math::gcd(strideW, dilationW);
+
+    auto yTilda = dilationH / gcdStrideDilationH;
+    auto xTilda = dilationW / gcdStrideDilationW;
+
+    auto yDot = math::integer_divide_ceil(y, yTilda);
+    auto xDot = math::integer_divide_ceil(x, xTilda);
+
+    auto hTilda = ho + math::integer_divide_ceil(dilationH * (y - 1), strideH);
+    auto wTilda = wo + math::integer_divide_ceil(dilationW * (x - 1), strideW);
+
+    auto iHTildaLeft = math::integer_divide_floor(
+        std::max(0l, leftPadH - dilationH * (yTilda - 1)), strideH);
+    auto iWTildaLeft = math::integer_divide_floor(
+        std::max(0l, leftPadW - dilationW * (xTilda - 1)), strideW);
+
+    auto iHTildaRight = std::min(
+        hTilda, math::integer_divide_ceil(leftPadH + hi - 1, strideH) + 1);
+    auto iWTildaRight = std::min(
+        wTilda, math::integer_divide_ceil(leftPadW + wi - 1, strideW) + 1);
+
+    auto hTildaSlice = iHTildaRight - iHTildaLeft;
+    auto wTildaSlice = iWTildaRight - iWTildaLeft;
+
+    auto gemm_id = 0;
```

Comment:
`gemm_id` -> `gemmId` to conform with naming convention. same with variables below with `_`.